internal/translation/networkpeering/conversion.go

```diff
@@ -0,0 +1,125 @@
+package networkpeering
+
+import (
+	"fmt"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/provider"
+)
+
+type ProviderContainer struct {
+	ID             string
+	ProviderName   provider.ProviderName
+	AtlasCIDRBlock string
+	RegionName     string // AWS
+	Region         string // Azure
+}
+
+func toAtlasConnection(peer *akov2.NetworkPeer) *admin.BaseNetworkPeeringConnectionSettings {
+	switch peer.ProviderName {
+	case provider.ProviderAWS:
+		return &admin.BaseNetworkPeeringConnectionSettings{
+			ContainerId:         peer.ContainerID,
+			ProviderName:        pointer.SetOrNil(string(peer.ProviderName), ""),
+			AccepterRegionName:  pointer.SetOrNil(peer.AccepterRegionName, ""),
+			AwsAccountId:        pointer.SetOrNil(peer.AWSAccountID, ""),
+			RouteTableCidrBlock: pointer.SetOrNil(peer.RouteTableCIDRBlock, ""),
+			VpcId:               pointer.SetOrNil(peer.VpcID, ""),
+		}
+	case provider.ProviderGCP:
+		return &admin.BaseNetworkPeeringConnectionSettings{
+			ContainerId:  peer.ContainerID,
+			ProviderName: pointer.SetOrNil(string(peer.ProviderName), ""),
+			GcpProjectId: pointer.SetOrNil(peer.GCPProjectID, ""),
+			NetworkName:  pointer.SetOrNil(peer.NetworkName, ""),
+		}
+	case provider.ProviderAzure:
+		return &admin.BaseNetworkPeeringConnectionSettings{
+			ContainerId:         peer.ContainerID,
+			ProviderName:        pointer.SetOrNil(string(peer.ProviderName), ""),
+			AzureDirectoryId:    pointer.SetOrNil(peer.AzureDirectoryID, ""),
+			AzureSubscriptionId: pointer.SetOrNil(peer.AzureSubscriptionID, ""),
+			ResourceGroupName:   pointer.SetOrNil(peer.ResourceGroupName, ""),
+			VnetName:            pointer.SetOrNil(peer.VNetName, ""),
+		}
+	default:
+		panic(fmt.Errorf("unsupported provider %q", peer.ProviderName))
```

Comment:
i am sceptical this should panic 🤔 are we really considering this being a programmatic error?

Reply:
I am on the fence as well. There are 2 ways to hit this:

1. Our validation fails to filter the provider name in the CRD.
2. Atlas sends an incorrect value.

For the first one we could panic, is our fault. But for the second, I guess we should error out instead.

Reply:
@josvazg given this is driven by user input I am in strong favor of returning an error.

A panic is valid if for we wire things incorrectly internally, i.e. use the "wrong indexer name" or something like that. But anything that arrives "via wire" from the user to AKO shouldn't panic unless there are very very strong reasons.

Reply:
> Atlas sends an incorrect value.

same rule as above. This is Atlas output, not an internal wiring error, it shouldn't panic.

Reply:
I will introduce an error return value and change all the code accordingly.

Reply:
Done, we use errors both ways now.